spec/polaris-management-service.yml

```diff
@@ -1089,6 +1089,10 @@ components:
         clientId:
           type: string
           description: The output-only OAuth clientId associated with this principal if applicable
+        federated:
```

Comment:
This part LGTM, but it should go to a ML vote

Reply:
+1 to ML discussion.

What's the use case for exposing federated principal in the Polaris Management API? Acting as a proxy in front of an IdP sounds questionable to me.

Reply:
The main case is to support adding federated roles so that privileges can be defined. Per the original design doc, federated identities are created on the fly when a user logs in, but if we don't allow creation of federated roles, we can't define any privileges for those users until they've logged in. That makes things hard for the admins, IMO

Reply:
I updated the spec to be more clear about when federated principals are expected as return types/arguments. This unfortunately cluttered the generated code more than I had hoped. I'll post a link on the mailing list re: this change.

Reply:
Thanks for the clarification! The description of the property LGTM. However, I'm not not sure about exposing federated Principals in the Polaris API.

Granted, we have to expose federated Principal Roles in order to allow assignment to Catalog Roles, but do Principals have to be exposed?

Reply:
As @eric-maynard suggested, it might be more efficient to review the ML / doc discussion. GH comment threads are less convenient for long discussions.